spec/unit/registry_spec.rb

```diff
@@ -100,6 +100,14 @@ def do_register
       expect(foo[:prefix_one]).to eq function
     end
 
+    it 'allows to overwrite function arguments' do
```

Comment:
Not sure how arguments are being overwritten here, this test looks like it would pass prior to the changes?